phpunit/blocks/block-navigation-link-variations-test.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Navigation block rendering tests.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ */
+
+/**
+ * Tests for the Navigation block variations for post types.
+ *
+ * @group blocks
+ */
+class Block_Navigation_Link_Variations_Test extends WP_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+		register_post_type(
+			'custom_book',
+			array(
+				'labels'            => array(
+					'item_link' => 'Custom Book',
+				),
+				'public'            => true,
+				'show_in_rest'      => true,
+				'show_in_nav_menus' => true,
+			)
+		);
+		register_taxonomy(
+			'book_type',
+			'custom_book',
+			array(
+				'labels'            => array(
+					'item_link' => 'Book Type',
+				),
+				'show_in_nav_menus' => true,
+			)
+		);
+	}
+
+	public function tear_down() {
+		unregister_post_type( 'custom_book' );
+		unregister_taxonomy( 'book_type' );
+		parent::tear_down();
+	}
+
+	/**
+	 * @covers ::register_block_core_navigation_link_post_type_variation
+	 */
+	public function test_navigation_link_variations_custom_post_type() {
+		$registry       = WP_Block_Type_Registry::get_instance();
+		$nav_link_block = $registry->get_registered( 'core/navigation-link' );
+		$this->assertFalse( empty( $nav_link_block->variations ) );
```

Comment:
```suggestion
		$this-> assertNotEmpty( $nav_link_block->variations );
```

I think we can use this. See

https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions